img_proof/ipa_gce.py

```diff
@@ -22,21 +22,58 @@
 
 import json
 import os
+import time
+
+from contextlib import contextmanager, suppress
 
 from img_proof import ipa_utils
 from img_proof.ipa_constants import (
     GCE_DEFAULT_TYPE,
     GCE_DEFAULT_USER
 )
-from img_proof.ipa_exceptions import GCECloudException
-from img_proof.ipa_exceptions import GCECloudRetryableError
+from img_proof.ipa_exceptions import GCECloudException, IpaRetryableError
 from img_proof.ipa_cloud import IpaCloud
 
-from libcloud.common.google import ResourceNotFoundError
-from libcloud.common.google import GoogleBaseError
-from libcloud.common.google import QuotaExceededError
-from libcloud.compute.types import Provider
-from libcloud.compute.providers import get_driver
+from google.oauth2 import service_account
+from googleapiclient import discovery
+
+
+def get_message_from_http_error(error, resource_name):
+    """
+    Attempt to parse error message from json.
+
+    If there is an error get message use a default message string.
```

Comment:
?? I do not understand what this part of the doc string is trying to tell me.

Reply:
Well that makes two of us :wink: .